boards/rp-pico/examples/pico_usb_serial_interrupt.rs

```diff
@@ -160,48 +193,42 @@ fn main() -> ! {
 /// knowing nothing about USB.
 #[allow(non_snake_case)]
 #[interrupt]
-unsafe fn USBCTRL_IRQ() {
-    use core::sync::atomic::{AtomicBool, Ordering};
-
-    /// Note whether we've already printed the "hello" message.
-    static SAID_HELLO: AtomicBool = AtomicBool::new(false);
-
+fn USBCTRL_IRQ() {
     // Grab the global objects. This is OK as we only access them under interrupt.
-    let usb_dev = USB_DEVICE.as_mut().unwrap();
-    let serial = USB_SERIAL.as_mut().unwrap();
-
-    // Say hello exactly once on start-up
-    if !SAID_HELLO.load(Ordering::Relaxed) {
-        SAID_HELLO.store(true, Ordering::Relaxed);
-        let _ = serial.write(b"Hello, World!\r\n");
-    }
-
-    // Poll the USB driver with all of our supported USB Classes
-    if usb_dev.poll(&mut [serial]) {
-        let mut buf = [0u8; 64];
-        match serial.read(&mut buf) {
-            Err(_e) => {
-                // Do nothing
-            }
-            Ok(0) => {
-                // Do nothing
-            }
-            Ok(count) => {
-                // Convert to upper case
-                buf.iter_mut().take(count).for_each(|b| {
-                    b.make_ascii_uppercase();
-                });
-
-                // Send back to the host
-                let mut wr_ptr = &buf[..count];
-                while !wr_ptr.is_empty() {
-                    let _ = serial.write(wr_ptr).map(|len| {
-                        wr_ptr = &wr_ptr[len..];
+    do_with_usb(|usb| {
+        // Poll the USB driver with all of our supported USB Classes
+        if usb.usb_device.poll(&mut [&mut usb.usb_serial]) {
+            let mut buf = [0u8; 64];
+            match usb.usb_serial.read(&mut buf) {
+                Err(_e) => {
+                    // Do nothing
+                }
+                Ok(0) => {
+                    // Do nothing
+                }
+                Ok(count) => {
+                    // Convert to upper case
+                    buf.iter_mut().take(count).for_each(|b| {
+                        b.make_ascii_uppercase();
                     });
+
+                    // Send back to the host
+                    let mut wr_ptr = &buf[..count];
+                    while !wr_ptr.is_empty() {
+                        let _ = usb.usb_serial.write(wr_ptr).map(|len| {
+                            wr_ptr = &wr_ptr[len..];
+                        });
+                    }
                 }
             }
         }
-    }
+    });
+}
+
+#[allow(non_snake_case)]
+#[exception]
+fn SysTick() {
+    do_with_usb(|usb| usb.usb_device.poll(&mut [&mut usb.usb_serial]));
```

Comment:
A comment like "Keeps the usb_device updated in case the usb interrupt is not called often." or a better explanation why this is advisable might be useful here for people like me who only have a superficial understanding of embedded software development.